src/ui/ResponsiveComponents/ResponsiveTabs.ts

```diff
@@ -248,24 +255,36 @@ export class ResponsiveTabs implements IResponsiveComponent {
   }
 
   private bindDropdownContentEvents() {
+    this.dropdownClickListener = () => {
+      if (this.isDropdownOpen()) {
+        this.ignoreNextClick = true;
+      }
+    };
+
     this.documentClickListener = event => {
-      if (Utils.isHtmlElement(event.target)) {
-        const eventTarget = $$(<HTMLElement>event.target);
-        if (
-          !eventTarget.closest('coveo-tab-list-container') &&
-          !eventTarget.closest(ResponsiveTabs.TAB_IN_DROPDOWN_HEADER_CSS_CLASS) &&
-          !eventTarget.closest(ResponsiveTabs.TAB_IN_DROPDOWN_CSS_CLASS)
-        ) {
-          this.closeDropdown();
-        }
+      if (!this.ignoreNextClick) {
```

Comment:
I would rename the variable `ignoreNextDocumentClick` to indicate what listener logic it will impact.